test/gtest/frontend_test.cpp

```diff
@@ -168,4 +169,39 @@ TEST_F(P4CFrontendEnumValidation, InvalidType) {
     ASSERT_TRUE(errors.contains("type-declared types"));
 }
 
+// Tests for MoveInitializers
+struct P4CFrontendMoveInitializers : P4CFrontend {
+    P4CFrontendMoveInitializers() {
+        addPasses({new P4::ResolveReferences(&refMap),
+                   new P4::MoveInitializers(&refMap)});
+    }
+
+    P4::ReferenceMap refMap;
+};
+
+TEST_F(P4CFrontendMoveInitializers, P4ControlSrcInfo) {
+    std::string program = P4_SOURCE(R"(
+        control m() {
+            bool blah = false;
+            apply{}
+        }
+    )");
+    RedirectStderr errors;
+    const auto *prog = parseAndProcess(program);
+    errors.dumpAndReset();
```

Comment:
You don't need to redirect stderr since it is not inspeced.
```suggestion
    const auto *prog = parseAndProcess(program);

Reply:
Good catch. I had copy/pasted code from the other tests.